internal/provider/webhook_recipient_resource.go

```diff
@@ -145,6 +152,30 @@ func (*webhookRecipientResource) Schema(_ context.Context, _ resource.SchemaRequ
 					},
 				},
 			},
+			"header": schema.SetNestedBlock{
+				Description: "Custom headers for webhooks",
+				Validators: []validator.Set{
+					setvalidator.SizeAtMost(5),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"name": schema.StringAttribute{
+							Required:    true,
+							Description: "The name or key for the header",
+							Validators: []validator.String{
+								stringvalidator.LengthBetween(1, 64),
+							},
+						},
+						"value": schema.StringAttribute{
+							Description: "Value for the header",
+							Optional:    true,
+							Validators: []validator.String{
+								stringvalidator.LengthAtMost(256),
```

Comment:
just matching the discussion above :)
```suggestion
								stringvalidator.LengthAtMost(512),
```